src/Compilers/CSharp/Test/Semantic/Semantics/FunctionPointerTests.cs

```diff
@@ -1918,5 +1942,42 @@ void M(delegate*<void> ptr)
                 Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "ToString").WithArguments("delegate*<void>", "ToString").WithLocation(6, 13)
             );
         }
+
+        [Fact]
+        public void ArglistCannotBeUsedWithFunctionPointers()
+        {
+            var comp = CreateCompilationWithFunctionPointers(@"
+unsafe class C
+{
+    static void M(delegate*<string, int, void> ptr1, delegate*<__arglist, void> ptr2)
+    {
+        ptr1(__arglist(string.Empty, 1), 1);
+        ptr1(null, __arglist(string.Empty, 1));
+        ptr1(null, 1, __arglist(string.Empty, 1));
+        ptr2(__arglist(1, 2, 3, ptr1));
+    }
+}");
+
+            comp.VerifyDiagnostics(
+                // (4,64): error CS1031: Type expected
+                //     static void M(delegate*<string, int, void> ptr1, delegate*<__arglist, void> ptr2)
+                Diagnostic(ErrorCode.ERR_TypeExpected, "__arglist").WithLocation(4, 64),
+                // (4,64): error CS1003: Syntax error, ',' expected
+                //     static void M(delegate*<string, int, void> ptr1, delegate*<__arglist, void> ptr2)
+                Diagnostic(ErrorCode.ERR_SyntaxError, "__arglist").WithArguments(",", "__arglist").WithLocation(4, 64),
+                // (6,14): error CS1503: Argument 1: cannot convert from '__arglist' to 'string'
+                //         ptr1(__arglist(string.Empty, 1), 1);
+                Diagnostic(ErrorCode.ERR_BadArgType, "__arglist(string.Empty, 1)").WithArguments("1", "__arglist", "string").WithLocation(6, 14),
+                // (7,20): error CS1503: Argument 2: cannot convert from '__arglist' to 'int'
+                //         ptr1(null, __arglist(string.Empty, 1));
+                Diagnostic(ErrorCode.ERR_BadArgType, "__arglist(string.Empty, 1)").WithArguments("2", "__arglist", "int").WithLocation(7, 20),
+                // (8,9): error CS8756: Function pointer 'delegate*<string,int,void>' does not take 3 arguments
+                //         ptr1(null, 1, __arglist(string.Empty, 1));
```

Comment:
>ptr1(null, 1, __arglist(string.Empty, 1)); [](start = 27, length = 42)

Do we have a test for this scenario when pointer coming from metadata is VARARG?